modules/tensor_mechanics/src/materials/ComputeCappedWeakPlaneStress.C

```diff
@@ -0,0 +1,890 @@
+    _Jacobian_mult[_qp](1, 1, i, i) -= poisson_effect;
+    if (q_trial > 0.0)
+    {
+      _Jacobian_mult[_qp](2, 0, i, i) = _Jacobian_mult[_qp](0, 2, i, i) = trial20 / q_trial * _dq_dpt * dpt_depii;
+      _Jacobian_mult[_qp](2, 1, i, i) = _Jacobian_mult[_qp](1, 2, i, i) = trial21 / q_trial * _dq_dpt * dpt_depii;
+    }
+  }
+
+  const Real poisson_effect = - _elasticity_tensor[_qp](2, 2, 0, 0) / Ezzzz * (_dgaE_dqt * smoothed_q.dg[0] + gaE * smoothed_q.d2g[0][0] * _dp_dqt + gaE * smoothed_q.d2g[0][1] * _dq_dqt + gaE * smoothed_q.d2g_di[0][0] * (dintnl0_dqt + dintnl0_dq * _dq_dqt) + gaE * smoothed_q.d2g_di[0][1] * (dintnl1_dqt + dintnl1_dp * _dp_dqt + dintnl1_dq * _dq_dqt));
+
+  const Real dqt_dep20 = (q_trial == 0.0 ? 1.0 : trial20 / q_trial) * _elasticity_tensor[_qp](2, 0, 2, 0);
+  _Jacobian_mult[_qp](2, 2, 2, 0) = _Jacobian_mult[_qp](2, 2, 0, 2) = _dp_dqt * dqt_dep20;
+  _Jacobian_mult[_qp](0, 0, 2, 0) = _Jacobian_mult[_qp](0, 0, 0, 2) = _Jacobian_mult[_qp](1, 1, 2, 0) = _Jacobian_mult[_qp](1, 1, 0, 2) =  poisson_effect* dqt_dep20;
+  if (q_trial > 0.0)
+  {
+    // for q_trial=0, Jacobian_mult is just given by the elastic case
+    _Jacobian_mult[_qp](0, 2, 0, 2) = _Jacobian_mult[_qp](2, 0, 0, 2) = _Jacobian_mult[_qp](0, 2, 2, 0) = _Jacobian_mult[_qp](2, 0, 2, 0) = _elasticity_tensor[_qp](2, 0, 2, 0) * q / q_trial + trial20 * (_dq_dqt - q / q_trial) / q_trial * dqt_dep20;
+    _Jacobian_mult[_qp](1, 2, 0, 2) = _Jacobian_mult[_qp](2, 1, 0, 2) = _Jacobian_mult[_qp](1, 2, 2, 0) = _Jacobian_mult[_qp](2, 1, 2, 0) = trial21 * (_dq_dqt - q / q_trial) / q_trial * dqt_dep20;
+  }
+
+  const Real dqt_dep21 = (q_trial == 0.0 ? 1.0 : trial21 / q_trial) * _elasticity_tensor[_qp](2, 1, 2, 1);
+  _Jacobian_mult[_qp](2, 2, 2, 1) = _Jacobian_mult[_qp](2, 2, 1, 2) = _dp_dqt * dqt_dep21;
+  _Jacobian_mult[_qp](0, 0, 2, 1) = _Jacobian_mult[_qp](0, 0, 1, 2) = _Jacobian_mult[_qp](1, 1, 2, 1) = _Jacobian_mult[_qp](1, 1, 1, 2) = poisson_effect * dqt_dep21;
+  if (q_trial > 0.0)
+  {
+    // for q_trial=0, Jacobian_mult is just given by the elastic case
+    _Jacobian_mult[_qp](0, 2, 1, 2) = _Jacobian_mult[_qp](2, 0, 1, 2) = _Jacobian_mult[_qp](0, 2, 2, 1) = _Jacobian_mult[_qp](2, 0, 2, 1) = trial20 * (_dq_dqt - q / q_trial) / q_trial * dqt_dep21;
+    _Jacobian_mult[_qp](1, 2, 1, 2) = _Jacobian_mult[_qp](2, 1, 1, 2) = _Jacobian_mult[_qp](1, 2, 2, 1) = _Jacobian_mult[_qp](2, 1, 2, 1) = _elasticity_tensor[_qp](2, 1, 2, 1) * q / q_trial + trial21 * (_dq_dqt - q / q_trial) / q_trial * dqt_dep21;
+  }
+}
+
+void
+ComputeCappedWeakPlaneStress::unsmoothedYieldFunctions(Real p, Real q, const std::vector<Real> & intnl, std::vector<Real> & yf) const
+{
+  mooseAssert(intnl.size() == 2, "ComputeCappedWeakPlaneStress: yieldFunctions called with intnl size " + Moose::stringify(intnl.size()));
+  mooseAssert(yf.size() == 3, "ComputeCappedWeakPlaneStress: yieldFunctions called with yf size " + Moose::stringify(yf.size()));
```

Comment:
mooseAssert is a macro that directly passes all arguments to a streaming operator. That means you don't have to use the '+' operator or `stringify`.